### PR TITLE
Add tests for grade command

### LINKED ISSUE #126: Add tests for grade command

### LINKED ISSUE #216: Check for valid ID in `JsonAdaptedAssignment`

### LINKED ISSUE #218: Rename `JsonAdaptedGrade`

### PR DESCRIPTION
**Pull request by marcusc55**
This PR only can be merged after merging #210.
Closes #126 
Also fixes #216 and fixes #218.

### CODE REVIEW

**Review by FergusMok — APPROVED** [review]

File: src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java [hunk 1]
```diff
@@ -39,19 +39,22 @@ public CommandResult execute(Model model) throws CommandException {
...
 
-        if (targetIndex.getZeroBased() >= assignmentList.size()) {
+        if (index.getOneBased() >= model.getAssignmentCounter()) {
             throw new CommandException(Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
         }
 
-        Assignment assignmentToDelete = assignmentList.get(targetIndex.getZeroBased());
-        model.deleteAssignment(assignmentToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_ASSIGNMENT_SUCCESS, assignmentToDelete));
+        Assignment asgToDelete = assignmentList.stream()
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java]: Should this part use the Model's `getAssignment` by ID method?

File: src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java [hunk 2]
```diff
@@ -24,10 +24,10 @@
 
     public static final String MESSAGE_DELETE_ASSIGNMENT_SUCCESS = "Deleted assignment: %1$s";
 
-    private final Index targetIndex;
+    private final Index index;
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java]: Can change the usage string (line 21) from "index" to "ID" (I can't comment on unchanged code sry)

File: src/main/java/seedu/edrecord/logic/commands/DeleteGradeCommand.java [hunk 3]
```diff
@@ -33,27 +33,25 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes a grade from the student "
             + "identified by the index number used in the displayed student list. \n"
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/logic/commands/DeleteGradeCommand.java]:
```suggestion
            + "identified by the index number used in the displayed student list. "
            + "Specify the assignment to delete from using the assignment ID displayed in the Assignment View. \n"
```

File: src/main/java/seedu/edrecord/logic/commands/DeleteGradeCommand.java [hunk 4]
```diff
@@ -33,27 +33,25 @@
...
-    public static final String MESSAGE_NO_SUCH_ASSIGNMENT = "There is no assignment with this name";
-    public static final String MESSAGE_NO_SUCH_GRADE = "There is no grade for this assignment";
+    public static final String MESSAGE_NO_SUCH_GRADE = "There is no grade for this assignment.";
 
     private final Index index;
-    private final Name name;
+    private final Index id;
 
     /**
      * @param index Index of the student to grade.
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/logic/commands/DeleteGradeCommand.java]:
```suggestion
     * @param index Index of the student to delete grade.
```

File: src/main/java/seedu/edrecord/logic/commands/DeleteGradeCommand.java [hunk 5]
```diff
@@ -33,27 +33,25 @@
...
-     * @param name  Name of the assignment to grade.
+     * @param id    ID of the assignment to grade.
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/logic/commands/DeleteGradeCommand.java]:
```suggestion
     * @param id    ID of the assignment to delete grade.
```

File: src/main/java/seedu/edrecord/logic/commands/DeleteGradeCommand.java [hunk 6]
```diff
@@ -71,9 +69,14 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_NO_MODULE_SELECTED);
         }
 
-        // Get assignment
-        Assignment assignment = model.searchAssignment(name)
-                .orElseThrow(() -> new CommandException(MESSAGE_NO_SUCH_ASSIGNMENT));
+        List<Assignment> assignmentList = model.getAssignmentList();
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/logic/commands/DeleteGradeCommand.java]: I think this part can use the `Model::getAssignment` method too.

File: src/main/java/seedu/edrecord/logic/commands/EditAssignmentCommand.java [hunk 7]
```diff
@@ -68,14 +68,18 @@ public CommandResult execute(Model model) throws CommandException {
         }
         List<Assignment> assignmentList = model.getSelectedModule().getValue().getAssignmentList();
 
-        if (index.getZeroBased() >= assignmentList.size()) {
+        if (index.getOneBased() >= model.getAssignmentCounter()) {
             throw new CommandException(Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
         }
 
-        Assignment asgToEdit = assignmentList.get(index.getZeroBased());
+        Assignment asgToEdit = assignmentList.stream()
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/logic/commands/EditAssignmentCommand.java]: This one has to update the usage string too.

File: src/main/java/seedu/edrecord/logic/commands/GradeCommand.java [hunk 8]
```diff
@@ -37,38 +37,37 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Assigns a grade to the student "
             + "identified by the index number used in the displayed student list. \n"
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/logic/commands/GradeCommand.java]:
```suggestion
            + "identified by the index number used in the displayed student list. "
            + "Specify the assignment to grade using the assignment ID displayed in the Assignment View. \n"
```

**Review by uyencfi — APPROVED** [review]
LGTM!

File: src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java (hunk 1 above)
Comment (marcusc55) [reply src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java]: Fixed in #255

File: src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java (hunk 2 above)
Comment (marcusc55) [reply src/main/java/seedu/edrecord/logic/commands/DeleteAssignmentCommand.java]: Fixed in #225

**Pull request merged**